contracts/REXLimitOrderManager.sol

```diff
@@ -0,0 +1,184 @@
+     * @param _streamRate The streaming rate for the order.
+     * @param _price The price at which the order is placed.
+     */
+    function createLimitOrder(
+        address _market,
+        bool _isInverted,
+        int96 _streamRate,
+        uint256 _price,
+        uint256 _ttl
+    ) external {
+        IREXUniswapV3Market market = IREXUniswapV3Market(_market);
+        ISuperToken token = ISuperToken(market.inputToken());
+
+        // check if address(this) is an operator for the user
+        require(token.isOperatorFor(address(this), msg.sender), "ACL");
+
+        bytes32 taskId = createGelatoTask(msg.sender, _market);
+
+        limitOrders[msg.sender][_market] = LimitOrder(
+            _isInverted,
+            _streamRate,
+            _price,
+            taskId,
+            false,
+            _ttl
+        );
+        emit LimitOrderCreated(
+            msg.sender,
+            _market,
+            _isInverted,
+            _streamRate,
+            _price
+        );
+    }
+
+    /**
+     *
+     * @notice Cancels a limit order.
+     * @dev The order is cancelled for the specified market, msg.sender is used for user's address.
+     * @param _market The address of the REX market where the order is placed for.
+     */
+    function cancelLimitOrder(address _market) public {
+        LimitOrder memory order = limitOrders[msg.sender][_market];
+        _cancelTask(order.taskId);
+    }
+
+    function createGelatoTask(
+        address _user,
+        address _market
+    ) internal returns (bytes32 taskId) {
+        ModuleData memory moduleData = ModuleData({
+            modules: new Module[](1),
+            args: new bytes[](1)
+        });
+
+        moduleData.modules[0] = Module.RESOLVER;
+
+        moduleData.args[0] = _resolverModuleArg(
+            address(this),
+            abi.encodeCall(this.checker, (_user, _market))
+        );
+
+        taskId = _createTask(
+            address(this),
+            abi.encode(this.updateUserStream, abi.encodePacked(_user, _market)),
+            moduleData,
+            ETH
+        );
+    }
+
+    /**
+     *
+     * @notice Updates the user's stream if order is in limit.
+     * @dev The user's stream is updated based on the current price of the market.
+     * @param _user The address of the user whose stream is updated.
+     * @param _market The address of the REX market where the order is placed for.
+     */
+    function updateUserStream(address _user, address _market) external {
+        LimitOrder memory order = limitOrders[_user][_market];
+
+        if (order.ttl < block.timestamp && order.executed == false) {
+            _cancelTask(order.taskId);
+        }
+
+        IREXUniswapV3Market market = IREXUniswapV3Market(_market);
+        ISuperToken token = ISuperToken(market.inputToken());
+        uint256 price = uint256(uint(market.getLatestPrice()));
+        order.executed = true;
+        if (price < order.price) {
```

Comment:
This logic is duplicated across `checker`, see my other comment about adding `flowRate` argument to `updateStream`